nova/virt/vmwareapi/volumeops.py

```diff
@@ -27,12 +27,25 @@
 from nova.i18n import _
 from nova.virt import driver
 from nova.virt.vmwareapi import constants
+from nova.virt.vmwareapi.session import StableMoRefProxy
 from nova.virt.vmwareapi import vm_util
 
 CONF = nova.conf.CONF
 LOG = logging.getLogger(__name__)
 
 
+class VolumeMoRefProxy(StableMoRefProxy):
+    def __init__(self, connection_info_data):
+        volume_ref_name = connection_info_data['volume']
+        ref = vutil.get_moref(volume_ref_name, 'VirtualMachine')
+        super(VolumeMoRefProxy, self).__init__(ref)
+        self._connection_info_data = connection_info_data
```

Comment:
Any reason for saving the whole `connection_info_data` instead of just the `volume_id` from it? Did you plan the persist the change back to the DB?

Reply:
> Any reason for saving the whole connection_info_data instead of just the volume_id from it

There is some redundancy in the `connection_info["data"]`, and I was considering to make use of it. The updated PR will.

>  Did you plan the persist the change back to the DB?

No, not really. The only time I expect the lookup to be triggered is during detachment, so the whole block-device-mapping will be deleted then anyway.

Reply:
> The only time I expect the lookup to be triggered is during detachment

Might be right inside Nova. The consistency checks are checking for the existence though. We have to remove that part from the checks and hope that nobody else is hijacking the BDMs from the outside e.g. to move volumes around ;)

Do you know if Cinder ever uses this information from it's volume-attachment and thus also needs these stable-moref patches?

Reply:
> Do you know if Cinder ever uses this information from it's volume-attachment and thus also needs these stable-moref patches?

No, good point, though. I'll check and see how it is done in cinder.